### PR TITLE
Arduino Due compatibility

### PR DESCRIPTION
**Pull request by Palakis**
Quite explanatory, I guess. These changes aim to make this library compatible with Arduino Due boards.
Tested it on two boards: Arduino Due (obviously) and Arduino Leonardo ETH (for backward compatibility testing). The sketch used for testing exposes 6 joysticks (each with 64 buttons and all analog channels enabled) and "animates" the buttons and axises of each in a predictable pattern.
Can confirm this works fine on both Due and Leonardo ETH.

### CODE REVIEW

**Comment by MHeironimus** [comment]
@Palakis Thanks for the update. I will try to review it this weekend (if not sooner).

**Comment by MHeironimus** [comment]
@Palakis I was able to review these changes this morning and they look great. This will be a nice enhancement to the library. Thank you for contributing.

I am going to have you make a few minor updates to your branch (see my other comments). Once those updates are made, I will run the code changes through my test suite and verify everything is still working (probably will not get to that until this weekend). If everything checks out, I will merge this into master.

Thanks again for your help with adding Arudino Duo support. I am excited to have the library supporting that board again.

File: Joystick/library.properties [hunk 1]
```diff
@@ -2,8 +2,8 @@ name=Joystick
 version=2.0.4
```
Comment (MHeironimus) [inline Joystick/library.properties]: @Palakis Can you update the version to 2.0.5 for me?

File: README.md [hunk 2]
```diff
@@ -213,6 +213,7 @@ Others have tested this library with the following boards:
 
 - [SparkFun Pro Micro](https://www.sparkfun.com/products/12640)
 - [RobotDyn Micro ATmega32U4-MU](http://robotdyn.com/catalog/boards/micro_atmega32u4_mu/) by [@Sylverzerom](https://github.com/Sylverzerom)
+- [Arduino Due](https://www.arduino.cc/en/Main/ArduinoBoardDue)
```
Comment (MHeironimus) [inline README.md]: @Palakis I want you to get credit for helping out on this. Add the following (if you want):

~~~
by [@Palakis](https://github.com/Palakis)
~~~

**Comment by Palakis** [comment]
@MHeironimus Thanks! Changes done.

**Pull request merged**